portfolio/src/main/webapp/script.js

```diff
@@ -86,209 +99,55 @@ function showCaption(element) {
 /*
  * Fetch the authentication status of the user from the server
  */
- function authStatus() {
+ function authenticateUser(isMapPage) {
```

Comment:
I think we can still make this more generic so that it can work with any section you want to require authentication, as this is dependent on comments & maps. The important parameters you'll need are the container (i.e. commentsDiv and mapDiv), the login url (i.e. login.url or login.mapUrl), and the toggle function (i.e.e toggleCommentsForm or toggleClickListener) such that it looks something like:
```suggestion
 function authenticateUser(sectionContainer, loginUrl, toggleListener) {
```
Instead of using isMapPage to see which container/url/toggle to use, you can use these variables assuming that they'll have the correct info passed in so that you don't need to worry about trying to get the correct info for each section. Then you can call authenticateUser with the corresponding values.

Reply:
That makes sense, and I had it implemented that way first. On the gallery page, I also want the authentication to work for both the map section and the comments section. If I made it generic like that, then on the gallery page I would have to call the authentication function twice, one for the comments parameters and again for the map. I avoided doing that because I thought it would slow down the page when not necessary. But would that be preferred over having the conditions in the method?

Reply:
Ah, I understand your concern is with calling fetch multiple times. We can still make this generic by breaking out the response logic into it's own helper function using the same ideas. Then in this function, you can check if the section exists and call the helper function with the appropriate values

Reply:
Got it, thanks